File: MainActivity.java
package com.example.savemethod;

import android.os.Bundle;
import android.view.View;
import android.widget.Button;
import android.widget.TextView;
import androidx.appcompat.app.AppCompatActivity;
import com.example.savemethod.R;
import java.io.IOException;
import java.io.InputStream;


public class MainActivity extends AppCompatActivity {
    TextView textView;
    Button button;
    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
        textView = findViewById(R.id.textView);
        button = findViewById(R.id.btnReadFile);
        button.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                String string = "";
                InputStream inputStream = null;
                try {
                    inputStream = getAssets().open("Quotes.txt");
                    int size = inputStream.available();
                    byte[] buffer = new byte[size];
                    inputStream.read(buffer);
                    string = new String(buffer);
                } catch (IOException e) {
                    e.printStackTrace();
                }
                textView.setText(string);
            }
        });
    }
}

File: MainActivity.kt
package com.example.saveprogress

import android.os.Bundle
import android.widget.EditText
import androidx.appcompat.app.AppCompatActivity

class MainActivity : AppCompatActivity() {

    private lateinit var score: EditText
    private lateinit var quizName : EditText

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.edittext)
        score = findViewById(R.id.edit1)
        quizName = findViewById(R.id.edit2)
    }

    override fun onResume() {
        super.onResume()
        val sh = getSharedPreferences("MySharedPref", MODE_PRIVATE)
        val result = sh.getInt("score", 0)
        val qName = sh.getString("quizName", "")
        score.setText(result.toString())
        quizName.setText(qName)
    }

    override fun onPause() {
        super.onPause()
        val sharedPreferences = getSharedPreferences("MySharedPref", MODE_PRIVATE)
        val myEdit = sharedPreferences.edit()
        myEdit.putInt("score", score.text.toString().toInt())
        myEdit.putString("quizName", quizName.text.toString())
        myEdit.apply()
    }

}
